rust/src/sip/log.rs

```diff
@@ -0,0 +1,58 @@
+/* Copyright (C) 2019 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+// written by Giuseppe Longo <giuseppe@glongo.it>
+
+use json::*;
+use sip::sip::{SIPState, SIPTransaction};
+
+#[no_mangle]
+pub extern "C" fn rs_sip_log_json(_state: &mut SIPState, tx: &mut SIPTransaction) -> *mut JsonT
+{
+    let js = Json::object();
+
+    match tx.request {
+        Some(ref req) => {
+            js.set_string("method", &format!("{}", req.method));
```

Comment:
(minor) Format is required for any of these.  `js.set_string("method", &req.method)` is a little cleaner.